app/controllers/openid_connect/logout_controller.rb

```diff
@@ -5,34 +5,35 @@ class LogoutController < ApplicationController
     include SecureHeadersConcern
     include FullyAuthenticatable
 
-    before_action :set_devise_failure_redirect_for_concurrent_session_logout, only: [:index]
+    before_action :set_devise_failure_redirect_for_concurrent_session_logout,
+                  only: [:logout]
     before_action :confirm_two_factor_authenticated, only: [:delete]
 
-    def index
+    # Handle logout (with confirmation if initiated by relying partner)
+    def logout
       @logout_form = build_logout_form
-
       result = @logout_form.submit
-      analytics.oidc_logout_requested(**result.to_h.except(:redirect_uri))
+      redirect_uri = result.extra[:redirect_uri]
+
+      analytics.oidc_logout_requested(**to_event(result))
 
-      if result.success? && result.extra[:redirect_uri]
-        handle_successful_logout_request(result, result.extra[:redirect_uri])
+      if result.success? && redirect_uri
+        handle_successful_logout_request(result, redirect_uri)
       else
         render :error
       end
     end
 
+    # Sign out without confirmation
```

Comment:
this is a super useful comment, thx for adding this context!

Reply:
I thought the `delete` route was the form submission to confirm logout?

Reply:
@mitchellhenke - It isn't! The `DELETE` route bypasses confirmation. This is more obvious if you look at [the view for this route](https://github.com/18F/identity-idp/blob/947d9a0c03a42bad5992674605e2e8ec5f5bf602/app/views/openid_connect/logout/confirm_logout.html.erb). 

The flow was effectively:

```
IF `GET` 
   THEN Do we need to confirm logout (this should always be true now)
       YES -> render index.html.erb
       NO -> Complete `DELETE` route steps
ELSE IF `DELETE`
   Call `ApplicationController#sign_out`
   AND
   Redirect the user to the partner provisioned redirect URI
```

The "new" route is the same as the old one and just handles both `GET` and `POST`.

Reply:
My understanding is that `DELETE` _is_ the confirmation rather than being something that skips it?

Reply:
ooo that might be better to clarify, maybe "does the logout, is the action after the confirmation is shown"

Reply:
yeah, sorry, that's what I was trying to get at.